File: problems/easy/binary-search/iterative.c
int search(int* nums, int numsSize, int target){
    int low = 0,
        high = numsSize - 1;
    
    while (low <= high) {
        int mid = (low + high) / 2;

        if (nums[mid] == target)
            return mid;
        else if (nums[mid] > target)
            high = mid - 1;
        else // if (nums[mid] > target)
            low = mid + 1;
    }

    return -1;
}

File: medium/search-a-2d-matrix/iterative.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        low = 0
        high = len(matrix) * len(matrix[0]) - 1

        while low <= high:
            mid = (high - low) // 2 + low
            i = mid // len(matrix[0])
            j = mid % len(matrix[0])
            if target > matrix[i][j]:
                low = mid + 1
            elif target < matrix[i][j]:
                high = mid - 1
            else:
                return True
        
        return False
